redash/handlers/query_results.py

```diff
@@ -9,14 +10,63 @@
 from redash.tasks import QueryTask, record_event
 from redash.permissions import require_permission, not_view_only, has_access, require_access, view_only
 from redash.handlers.base import BaseResource, get_object_or_404
-from redash.utils import collect_query_parameters, collect_parameters_from_request
+from redash.utils import collect_query_parameters, collect_parameters_from_request, gen_query_hash
 from redash.tasks.queries import enqueue_query
 
 
 def error_response(message):
     return {'job': {'status': 4, 'error': message}}, 400
 
 
+#
+# Run a parameterized query synchronously and return the result
+# DISCLAIMER: Temporary solution to support parameters in queries. Should be
+#             removed once we refactor the query results API endpoints and handling
+#             on the client side. Please don't reuse in other API handlers.
+#
+def run_query_sync(data_source, parameter_values, query_text, max_age=0):
+    query_parameters = set(collect_query_parameters(query_text))
```

Comment:
its the same as for the usual `run_query` as well: https://github.com/getredash/redash/pull/1524/files#diff-f734d32676ec526e17e5ae92d41b5787R71